### PR TITLE
Add and initialize cluster id in the TSO service

### PR DESCRIPTION
**Pull request by binshi-bing**
### What problem does this PR solve?

Issue Number: Ref #5836 

### What is changed and how does it work?

```commit-message
Add and initialize cluster id in the TSO service.
PD/API service discovery gets cluster id from the API service.
It then passes the cluster id to the TSO service because both share the same cluster id.
```

### Check List

Tests

- Unit test

### Release note

```release-note
None.
```

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -376,8 +384,8 @@ func (c *client) scheduleUpdateTokenConnection() {
 }
 
 // GetClusterID returns the ClusterID.
-func (c *client) GetClusterID(ctx context.Context) uint64 {
-	return c.svcDiscovery.GetClusterID(ctx)
+func (c *client) GetClusterID(context.Context) uint64 {
```
Comment (rleungx) [inline client/client.go:387]: Do we need the context?

Comment (binshi-bing) [reply client/client.go:387]: No need, but it's defined in the Client interfact. I can't change it.

type Client interface {
	// GetClusterID gets the cluster ID from PD.
	GetClusterID(ctx context.Context) uint64

File: client/client.go [hunk 2]
```diff
@@ -275,7 +275,7 @@ func NewClient(svrAddrs []string, security SecurityOption, opts ...ClientOption)
 // NewClientWithContext creates a PD client with context.
 func NewClientWithContext(ctx context.Context, svrAddrs []string, security SecurityOption, opts ...ClientOption) (Client, error) {
 	log.Info("[pd] create pd client with endpoints", zap.Strings("pd-address", svrAddrs))
-	c, clientCtx, clientCancel, tlsCfg := createClient(ctx, &security)
+	c, clientCtx, clientCancel, tlsCfg := createClient(ctx, 0, &security)
```
Comment (rleungx) [inline client/client.go:278]: Please add a comment to explain that `0` represents the default group.

**Review by rleungx — APPROVED** [review]
The rest LGTM

**Review by lhy1024 — APPROVED** [review]

File: client/client.go (hunk 2 above)
Comment (binshi-bing) [reply client/client.go:278]: added

**Pull request merged**